Makefile.cache

```diff
@@ -71,7 +71,9 @@ SONIC_COMMON_FLAGS_LIST :=  $(CONFIGURED_PLATFORM) \
                             $(CONFIGURED_ARCH) \
                             $(BLDENV) \
                             $(SONIC_DEBUGGING_ON) \
-                            $(SONIC_PROFILING_ON) $(SONIC_ENABLE_SYNCD_RPC)
+                            $(SONIC_PROFILING_ON) \
+                            $(SONIC_ENABLE_SYNCD_RPC) \
+                            $(ENABLE_ASAN)
```

Comment:
Where is the flag used? It is only a flag? why not use cache=none/wcache, if we want to disable/rebuild cache?

Reply:
It's used for building a sonic image with ASAN-enabled containers (currently, swss and mellanox-syncd are supported).
More details: https://github.com/Azure/sonic-buildimage/pull/9857
The idea is to have the same cache working (in rw mode) for both regular and ASAN-enabled builds.
Currently, it's not possible since there is no distinction between packages built with/without ASAN_ENABLE=y.

Reply:
Can we only enable it for special targets, such as only for docker-orchagent? It will make the all the targets with  ENABLE_ASAN flag not able to share with the ones without the flag. Is it expected?

Reply:
Ok, this will work as well. I've updated the PR.